collector/consumer/exporter/otelexporter/defaultadapter/label_converter.go

```diff
@@ -313,14 +313,25 @@ func (m *LabelConverter) transform(group *model.GaugeGroup) (*model.AttributeMap
 	return attrsMap, attrs.attrsMapPool.Free
 }
 
-// convert is used to general final labels for Sync Metric and Trace. It won't modify the origin model.GaugeGroup and should be free by calling the FreeAttrsList after exported.
-func (m *LabelConverter) convert(group *model.GaugeGroup) ([]attribute.KeyValue, FreeAttrsList) {
-	labels := group.Labels
+// searchExtraAttribute determine the final outPut struct for incoming gaugeGroup
+// return the default struct of `UNSUPPORTED Protocol` if failed
+func (m *LabelConverter) searchExtraAttribute(labels *model.AttributeMap) realAttributes {
 	tmpExtraKey := &extraLabelsKey{protocol: empty}
 	for i := 0; i < len(m.updateKeys); i++ {
 		tmpExtraKey = m.updateKeys[i](tmpExtraKey, labels)
 	}
-	attrs := m.labelsMap[*tmpExtraKey]
+	attrs, ok := m.labelsMap[*tmpExtraKey]
+	if !ok {
+		tmpExtraKey.protocol = UNSUPPORTED
+		attrs = m.labelsMap[*tmpExtraKey]
+	}
+	return attrs
```

Comment:
Better if we add a default result value when the `attrs` cannot be found even if using `UNSUPPORTED`. Since currently `UNSUPPORTED` always resides on the map, the modification could be made later.